Tests/RegexTests/AnyRegexOutputTests.swift

```diff
@@ -0,0 +1,120 @@
+
+import _StringProcessing
+import XCTest
+
```

Comment:
There are already some tests on `AnyRegexOutput` in `RegexDSLTests.testDynamicCaptures()`. I'd suggest moving those tests here now that we have a dedicated file.